awx/main/models/workflow.py

```diff
@@ -57,6 +57,20 @@
 logger = logging.getLogger('awx.main.models.workflow')
 
 
+def get_artifacts_from_unified_job(job):
```

Comment:
Instead of this standalone function, why not just have different implementations of `.get_combined_artifacts()` on the different `UnifiedJob` subtypes?

Reply:
Took me a while to process, but I wound up going with exactly this suggestion. That allowed getting rid of this method altogether.